solr/webapp/web/js/angular/controllers/query.js

```diff
@@ -165,13 +165,21 @@ solrAdminApp.controller('QueryController',
                   return param.indexOf(argParam) === 0;
               });
       }
+
+      var getHighlighterFieldsToPurge = function(hlMethod){
+          // first, select the fieldsets which ng-show includes val['hl.method'] but not val['hl.method']==${hlMethod}
+          // then, select their descendants that have ng-model
+          const selector = `div.fieldset[ng-show*="val['hl.method']"]:not([ng-show*="val['hl.method']=='${hlMethod}'"])
```

Comment:
if this works, great..   it's pretty gnarly css code!  I appreciate the comment explaining it ;-)

Reply:
I cannot think of other more elegant way to handle this. Agree this is not the best as it also makes assumptions about the `ng-show` expression and `ng-model` value. Open to hear and try out other suggestions.